build-logic/packaging/src/main/kotlin/gradlebuild/instrumentation/transforms/InstrumentationMetadataTransform.kt

```diff
@@ -95,7 +95,8 @@ abstract class InstrumentationMetadataTransform : TransformAction<TransformParam
     private
     fun handleClassChange(change: FileChange, superTypes: Properties) {
         val className = change.normalizedPath.removeSuffix(".class")
-        when (change.changeType) {
+        @Suppress("UNNECESSARY_NOT_NULL_ASSERTION")
+        when (change.changeType!!) {
```

Comment:
Is this a Kotlin bug? Was it reported?

Reply:
This is indeed a bug in Kotlin that is present in 2.0.21 that we currently use.
That bug is already fixed in Kotlin 2.1.0.
I reworked the commit message and added a comment.